tests/core/test_grid.py

```diff
@@ -811,82 +811,118 @@ def test_cast_rectilinear_grid():
         assert grid.origin == origin
         assert grid.spacing == spacing
 
     # just dims (deprecated)
-    with pytest.warns(
-        PyVistaDeprecationWarning, match="Behavior of pyvista.UniformGrid has changed"
-    ):
-        grid = pyvista.UniformGrid(dims)
+    with pytest.warns(PyVistaDeprecationWarning, match="Behavior of pyvista.ImageData has changed"):
+        grid = pyvista.ImageData(dims)
         assert grid.dimensions == dims
 
     with pytest.warns(
         PyVistaDeprecationWarning,
         match='`dims` argument is deprecated. Please use `dimensions`.',
     ):
-        grid = pyvista.UniformGrid(dims=dims)
+        grid = pyvista.ImageData(dims=dims)
     with pytest.raises(TypeError):
-        grid = pyvista.UniformGrid(dimensions=dims, dims=dims)
+        grid = pyvista.ImageData(dimensions=dims, dims=dims)
 
     # uniform grid from a uniform grid
-    grid = pyvista.UniformGrid(dimensions=dims, spacing=spacing, origin=origin)
-    grid_from_grid = pyvista.UniformGrid(grid)
+    grid = pyvista.ImageData(dimensions=dims, spacing=spacing, origin=origin)
+    grid_from_grid = pyvista.ImageData(grid)
     assert grid == grid_from_grid
 
     # and is a copy
     grid.origin = (0, 0, 0)
     assert grid != grid_from_grid
 
 
-def test_uniform_grid_invald_args():
-    with pytest.warns(
-        PyVistaDeprecationWarning, match="Behavior of pyvista.UniformGrid has changed"
-    ):
-        pyvista.UniformGrid((1, 1, 1))
+def test_create_uniform_grid_from_specs():
+    # empty
+    with pytest.warns(PyVistaDeprecationWarning):
+        grid = pyvista.UniformGrid()
+
+    # create ImageData
+    dims = (10, 10, 10)
+    with pytest.warns(PyVistaDeprecationWarning):
+        grid = pyvista.UniformGrid(dimensions=dims)  # Using default spacing and origin
+    assert grid.dimensions == dims
+    assert grid.extent == (0, 9, 0, 9, 0, 9)
+    assert grid.origin == (0.0, 0.0, 0.0)
+    assert grid.spacing == (1.0, 1.0, 1.0)
+
+    # Using default origin
+    spacing = (2, 1, 5)
+    with pytest.warns(PyVistaDeprecationWarning):
+        grid = pyvista.UniformGrid(dimensions=dims, spacing=spacing)
+    assert grid.dimensions == dims
+    assert grid.origin == (0.0, 0.0, 0.0)
+    assert grid.spacing == spacing
+    origin = (10, 35, 50)
+
+    # Everything is specified
+    with pytest.warns(PyVistaDeprecationWarning):
+        grid = pyvista.UniformGrid(dimensions=dims, spacing=spacing, origin=origin)
+    assert grid.dimensions == dims
+    assert grid.origin == origin
+    assert grid.spacing == spacing
+
+    # uniform grid from a uniform grid
+    with pytest.warns(PyVistaDeprecationWarning):
+        grid = pyvista.UniformGrid(dimensions=dims, spacing=spacing, origin=origin)
+    with pytest.warns(PyVistaDeprecationWarning):
+        grid_from_grid = pyvista.UniformGrid(grid)
+    assert grid == grid_from_grid
+
+    # and is a copy
+    grid.origin = (0, 0, 0)
+    assert grid != grid_from_grid
```

Comment:
In my mind, it is sufficient to test UniformGrid to verify that a deprecated warning is generated when called.
```suggestion
def test_create_uniform_grid():
    with pytest.warns(PyVistaDeprecationWarning):
        grid = pyvista.UniformGrid()
```

Reply:
Sufficient, yes... but... My original implementation could be reduced but I do want to make sure that arguments are passed along safely. Perhaps the `UniformGrid` class's `__init__` should just pass `*args. **kwargs` instead of listing out the same function signature